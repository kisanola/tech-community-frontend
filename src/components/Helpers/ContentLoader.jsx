import React from 'react';
import ReactContentLoader from 'react-content-loader';
import PropTypes from 'prop-types';

const ContentLoader = ({ style, count }) => [
  ...Array(count || 1)].map((value, index) => (
    // eslint-disable-next-line react/no-array-index-key
    <div style={{ ...style, width: '100%' }} key={index}>
      <ReactContentLoader
        height={130}
        width={400}
        speed={2}
        primaryColor="#fff"
        secondaryColor="#e7f1fa"
      >
        <rect x="50" y="10" rx="4" ry="4" width="117" height="6" />
        <rect x="50" y="25" rx="3" ry="3" width="85" height="6" />
        <rect x="0" y="60" rx="3" ry="3" width="350" height="6" />
        <rect x="0" y="80" rx="3" ry="3" width="380" height="6" />
        <rect x="0" y="100" rx="3" ry="3" width="380" height="6" />
        <circle cx="20" cy="20" r="20" />
      </ReactContentLoader>
    </div>
));

ContentLoader.propTypes = {
  style: PropTypes.object,
};

ContentLoader.defaultProps = {
  style: {},
};
export default ContentLoader;
