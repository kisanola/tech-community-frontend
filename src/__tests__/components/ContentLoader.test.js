import React from 'react';
import ContentLoader from '../../components/Helpers/ContentLoader';
import { mount } from 'enzyme';

describe('ContentLoader', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<ContentLoader />);
  });

  it('should render `ContentLoader`', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
