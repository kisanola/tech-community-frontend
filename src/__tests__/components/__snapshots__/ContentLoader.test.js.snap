// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoader should render \`ContentLoader\` 1`] = `ReactWrapper {}`;
